From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nkpro2000 <45198561+nkpro2000@users.noreply.github.com>
Date: Mon, 19 Jun 2023 17:50:46 +0530
Subject: [PATCH] Apply NoRoot for Flatpak system-helper, while building pkg

This patch is used to modify PKGBUILD of arch flatpak pkg, to apply my modifications
on flatpak_git/system-helper/flatpak-system-helper.c while `makepkg -Lfs`.
This file is generated using `git format-patch --zero-commit -1 HEAD`.
---
 PKGBUILD | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/PKGBUILD b/PKGBUILD
index 3583647..a40c701 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -67,6 +67,14 @@ b2sums=('SKIP'
         '81010a49af57e6047ab57285e3f8430da3b2be8504bc5600502da9a3729e9f92bb086a314ca8a03bca446e8f203150e378f3443f7034d8d7ade9dad6f1b9db59'
         '1c45caa65e2a1598f219977d5a81dcb8ea5d458880c43c40ba452b0c77cbbf41b36fa6911741f22c807d318e04e39e4fcc1455ed8d68faaba10162dae2570abc')
 
+source+=(Using-flatpak-chown_noroot-instead-of-unistd-chown-so-noroot.patch
+         Using-revokefs-fuse_noroot-instead-of-unistd-setugid-so-noroot.patch
+         Providing-necessary-permissions-on-paths-so-no-need-root.patch)
+b2sums+=('SKIP'
+         'SKIP'
+         'SKIP')
+
+
 pkgver() {
   cd flatpak
   git describe --tags | sed 's/[^-]*-g/r&/;s/-/+/g'
@@ -83,6 +91,11 @@ prepare() {
   # https://bugs.archlinux.org/task/75623
   git apply -3 ../fusermount3.diff
 
+  # Do NoRoot modifications in flatpak/system-helper/flatpak-system-helper.c
+  git apply -3 ../Using-flatpak-chown_noroot-instead-of-unistd-chown-so-noroot.patch
+  git apply -3 ../Using-revokefs-fuse_noroot-instead-of-unistd-setugid-so-noroot.patch
+  git apply -3 ../Providing-necessary-permissions-on-paths-so-no-need-root.patch
+
   git submodule init
   git submodule set-url libglnx "$srcdir/libglnx"
   git submodule set-url bubblewrap "$srcdir/bubblewrap"
-- 
2.40.1

